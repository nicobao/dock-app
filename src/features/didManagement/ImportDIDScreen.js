import {
  BackButton,
  Box,
  Header,
  Input,
  LoadingButton,
  NBox,
  ScreenContainer,
} from '../../design-system';
import {addTestId} from '../../core/automation-utils';
import {navigateBack} from '../../core/navigation';
import {translate} from '../../locales';
import {FormControl, ScrollView, Stack} from 'native-base';
import React, {useCallback} from 'react';
import {useDIDManagementHandlers} from './didHooks';

export function ImportDIDScreen({
  form,
  handleChange,
  handleSubmit,
  encryptedJSONWallet,
}) {
  return (
    <ScreenContainer {...addTestId('ImportDIDScreen')}>
      <Header>
        <Box
          marginLeft={1}
          marginRight={22}
          flexDirection="row"
          alignItems="center">
          <NBox width={'80px'}>
            <BackButton {...addTestId('GoBack')} onPress={navigateBack} />
          </NBox>
          <NBox
            flex={1}
            width="100%"
            alignContent="center"
            alignItems="center"
            pl={15}
          />
          <NBox width="80px" alignItems="flex-end" />
        </Box>
      </Header>
      <ScrollView marginLeft={5} marginRight={5}>
        <FormControl>
          <Stack mt={7}>
            <FormControl.Label>
              {translate('password_input.title')}
            </FormControl.Label>
            <Input
              {...addTestId('Password')}
              value={form.password}
              onChangeText={handleChange('password')}
              autoCapitalize="none"
              secureTextEntry={true}
            />
          </Stack>
        </FormControl>
      </ScrollView>
      <NBox mx={7}>
        <LoadingButton
          {...addTestId('Next')}
          full
          testID="next-btn"
          mb={70}
          onPress={handleSubmit}
          isDisabled={
            form.password.trim().length <= 0 && !!encryptedJSONWallet
          }>
          {translate('password_input.submit')}
        </LoadingButton>
      </NBox>
    </ScreenContainer>
  );
}

export function ImportDIDScreenContainer({route}) {
  const {encryptedJSONWallet} = route.params || {};
  const {form, handleChange, onImportDID} = useDIDManagementHandlers();

  const handleSubmit = useCallback(async () => {
    await onImportDID({encryptedJSONWallet, password: form.password});
  }, [encryptedJSONWallet, form.password, onImportDID]);
  return (
    <ImportDIDScreen
      form={form}
      handleChange={handleChange}
      handleSubmit={handleSubmit}
      encryptedJSONWallet={encryptedJSONWallet}
    />
  );
}
