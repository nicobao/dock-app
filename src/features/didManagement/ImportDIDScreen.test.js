import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {ImportDIDScreen} from './ImportDIDScreen';

const mockStore = configureMockStore();

describe('ImportDIDScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const form = {
      didName: '',
      didType: '',
      password: '',
      showDIDDockQuickInfo: true,
      keypairType: 'ed25519',
      derivationPath: '',
      _errors: {
        didType: '',
      },
      _hasError: false,
    };
    const encryptedJSONWallet = {};
    const wrapper = shallow(
      <ImportDIDScreen
        form={form}
        handleChange={jest.fn()}
        handleSubmit={jest.fn()}
        encryptedJSONWallet={encryptedJSONWallet}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
