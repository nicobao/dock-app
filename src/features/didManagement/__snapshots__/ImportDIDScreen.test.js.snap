// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportDIDScreen should render correctly 1`] = `
<RNCSafeAreaView
  accessibilityLabel="ImportDIDScreen"
  style={
    Object {
      "backgroundColor": "#141417",
      "flex": 1,
    }
  }
  testID="ImportDIDScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <Box
          alignItems="center"
          flexDirection="row"
          marginLeft={1}
          marginRight={22}
        >
          <Memo(ForwardRef(Box))
            width="80px"
          >
            <BackButton
              accessibilityLabel="GoBack"
              onPress={[MockFunction]}
              testID="GoBack"
            />
          </Memo(ForwardRef(Box))>
          <Memo(ForwardRef(Box))
            alignContent="center"
            alignItems="center"
            flex={1}
            pl={15}
            width="100%"
          />
          <Memo(ForwardRef(Box))
            alignItems="flex-end"
            width="80px"
          />
        </Box>
      </Styled(View)>
      <ForwardRef
        marginLeft={5}
        marginRight={5}
      >
        <Memo(ForwardRef(FormControl))>
          <Memo(ForwardRef(Stack))
            mt={7}
          >
            <Memo(ForwardRef(FormControlLabel))>
              Enter password
            </Memo(ForwardRef(FormControlLabel))>
            <Input
              accessibilityLabel="Password"
              autoCapitalize="none"
              secureTextEntry={true}
              testID="Password"
              value=""
            />
          </Memo(ForwardRef(Stack))>
        </Memo(ForwardRef(FormControl))>
      </ForwardRef>
      <Memo(ForwardRef(Box))
        mx={7}
      >
        <LoadingButton
          accessibilityLabel="Next"
          full={true}
          isDisabled={Object {}}
          mb={70}
          onPress={[MockFunction]}
          testID="next-btn"
        >
          Next
        </LoadingButton>
      </Memo(ForwardRef(Box))>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;
